frontend/src/components/DetailView/common/EditableTable.tsx

```diff
@@ -21,37 +21,53 @@ const getNewState = (state: RowState): RowState => {
   return 'clean'
 }
 
-export const EditableTable = <T extends MRT_RowData & { rowState?: RowState }, ParentType extends MRT_RowData>({
+export const EditableTable = <
+  T extends MRT_RowData & { rowState?: RowState; index?: number },
+  ParentType extends MRT_RowData,
+>({
   tableData,
   editTableData,
   columns,
   field,
-  visible_data,
+  visible_data, // use some filtered data instead of the actual data. Allows you to hide some rows. But be careful that the data is in the right format
+  useDefinedIndex = false, // Control whether to use the defined index or static index. The index data needs to have a key named 'index'.
+  useObject = false,
 }: {
   tableData?: Array<T> | null
   editTableData?: Array<EditDataType<T>> | null
   columns: MRT_ColumnDef<T>[]
   field: keyof EditDataType<ParentType>
   visible_data?: Array<T>
+  useDefinedIndex?: boolean
+  useObject?: boolean
 }) => {
   const [pagination, setPagination] = useState<MRT_PaginationState>(defaultPagination)
   const { editData, setEditData, mode, data } = useDetailContext<ParentType>()
+
   if (tableData === null || editTableData === null) return <CircularProgress />
 
   const actionRow = ({ row, staticRowIndex }: { row: MRT_Row<T>; staticRowIndex?: number | undefined }) => {
     const state = row.original.rowState ?? 'clean'
 
-    // TODO: Using static index - need to use some id, sorting breaks this
-    const rowClicked = (index: number | undefined) => {
+    // uses either 'row.original.index' or 'staticRowIndex' based on 'useDefinedIndex'
+    const rowClicked = () => {
+      let index = useDefinedIndex ? row.original.index : staticRowIndex
       if (index === undefined) return
-      const items = [...editData[field]]
+      let items: Array<EditDataType<T>>
+      if (useObject) {
+        items = [editData[field]]
+        index = 0
+      } else {
+        items = [...editData[field]]
+      }
+
       if (items[index].rowState === 'new') {
         items.splice(index, 1)
-        setEditData({ ...editData, [field]: items })
+        setEditData({ ...editData, [field]: useObject ? items[0] : items })
```

Comment:
This relates also to the `editData[field]` but is also items quaranteed to have at least one element?

Reply:
const ActionRow is only called when a row in the table is clicked. Which in theory should mean that there is at least a single element in items. But again not sure if this is sufficient

Reply:
Yeah this is reasonable. It would probably get caught in the `if (index === undefined) return` clause.